src/pages/buyer/Orders.tsx

```diff
@@ -0,0 +1,279 @@
+
+
+import { useState, useEffect, Fragment } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Dialog, Transition } from '@headlessui/react';
+import moment, { Moment } from 'moment'; 
+import { OrdersAction, getByIdOrdersAction } from '../../redux/action/OrdersAction';
+ import { Navigation} from "../../components";
+
+ 
+interface Product {
+  name: string | number | null | undefined;
+  quantity: number | null | undefined;
+  subtotal: number | null | undefined;
+}
+
+ 
+interface OrdersData {
+  data: any | null | undefined;  
```

Comment:
data inside the OrdersData interface can have first interface then other options like null and undefined